source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -1020,14 +1020,22 @@ void ShaderGraph::optimize(GenContext& context)
     size_t numEdits = 0;
     for (ShaderNode* node : getNodes())
     {
-        if (node->hasClassification(ShaderNode::Classification::CONSTANT) ||
-            node->hasClassification(ShaderNode::Classification::DOT))
+        if (node->hasClassification(ShaderNode::Classification::CONSTANT))
         {
-            // Constant and dot nodes can be removed by moving their value
-            // or connection downstream.
+            // Constant nodes can be elided by moving their value downstream.
             bypass(context, node, 0);
             ++numEdits;
         }
+        else if (node->hasClassification(ShaderNode::Classification::DOT))
+        {
+            // Dot nodes without modifiers can be elided by moving their connection downstream.
+            ShaderInput* in = node->getInput("in");
+            if (in->getChannels().empty())
```

Comment:
BTW, if we add this, then the type should be checked as validation fails on this, but this logic does not.
e.g. if it's a string. then compilation will fail.

Reply:
Can you clarify what you mean here?

Reply:
Maybe you're referring to the presence of an empty `channels` string on an `input` element in MaterialX?  That would indeed emit a validation warning, but this object is actually a `ShaderNode` within the shader generation system, which uses an empty `channels` string to mean no channels attribute is present.